FrontEnd/src/views/GradesComponent.js

```diff
@@ -0,0 +1,119 @@
+import React, {useEffect, useState} from 'react';
+import AccordionSummary from "@material-ui/core/AccordionSummary";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import Typography from "@material-ui/core/Typography";
+import AccordionDetails from "@material-ui/core/AccordionDetails";
+import Accordion from "@material-ui/core/Accordion";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+import {connect} from "react-redux";
+import {Table} from "@material-ui/core";
+import TableContainer from "@material-ui/core/TableContainer";
+import Paper from "@material-ui/core/Paper";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableCell from "@material-ui/core/TableCell";
+import TableBody from "@material-ui/core/TableBody";
+import Box from "@material-ui/core/Box";
+import {SERVER} from "../../index";
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        width: '100%',
+    },
+    accordion: {
+        width: '100%',
+        minWidth: 650,
+    },
+    heading: {
+        fontSize: theme.typography.pxToRem(15),
+        fontWeight: theme.typography.fontWeightRegular,
+    },
+    table: {
+
+    }
+}));
+
+function GradesAccordions(props) {
+    const[allTasksOfUser, setAllTasksOfUser] = useState([]);
+    const classes = useStyles();
+
+    let requestBody = JSON.stringify({
+        userId: props.userId,
+        request_token: props.request_token
+    });
+
+    useEffect(() => {
```

Comment:
Ist es state of the art useEffect für Netzwerkanfragen zu verwenden? Mir kommt es ein wenig hacky vor.